tests/DFE.FindInformationAcademiesTrusts.CypressTests/cypress/e2e/regression/trusts/trust-contact-page.cy.ts

```diff
@@ -1,23 +1,130 @@
 import trustContactsPage from "../../../pages/trusts/trustContactsPage";
+import commonPage from "../../../pages/commonPage";
 
 describe("Testing the components of the Trust contacts page", () => {
 
-    describe("On a Trust Overview page with data", () => {
+    describe("On a Trust contacts page with data", () => {
         beforeEach(() => {
             cy.login()
             cy.visit('/trusts/contacts?uid=5712')
         });
 
-        it("Can change contact", () => {
+        it("Can change Trust relationship manager contact details", () => {
             const randomNumber = Math.floor(Math.random() * 9999);
             const name = `Name${randomNumber}`
             const email = `emai${randomNumber}l@education.gov.uk`
-            trustContactsPage.editTRM(name, email);
-            trustContactsPage.elements.TrustRelationshipManager.Name().should('contain.text', name);
-            trustContactsPage.elements.TrustRelationshipManager.Email().should('contain.text', email);
-            cy.contains("Source and updates").click();
+
+            trustContactsPage
+                .checkTRMFieldsAndDatasource(name, email)
+
+        })
+
+        it("Can change Schools financial support oversight lead contact details", () => {
+            const randomNumber = Math.floor(Math.random() * 9999);
+            const name = `Name${randomNumber}`
+            const email = `emai${randomNumber}l@education.gov.uk`
+
+            trustContactsPage
+                .checkSFSOFieldsAndDatasource(name, email)
+        })
+
+        it("Checks a trusts external contact details", () => {
+            trustContactsPage
+                .checkAccountingOfficerPresent()
+                .checkChairOfTrusteesPresent()
+                .checkChiefFinancialOfficerPresent()
+        })
+
+    })
+
+    describe("On a different Trusts contacts page with data", () => {
+        beforeEach(() => {
+            cy.login()
+            cy.visit('/trusts/contacts?uid=5527')
+        });
+
+        it("Can change Trust relationship manager contact details", () => {
+            const randomNumber = Math.floor(Math.random() * 9999);
+            const name = `Name${randomNumber}`
+            const email = `emai${randomNumber}l@education.gov.uk`
```

Comment:
These are used all over the tests, can you move this into a separate function to generate the name and email?

Reply:
Also they all have the same spelling mistake I made when I put the test in 😂
```suggestion
            const randomNumber = Math.floor(Math.random() * 9999);
            const name = `Name${randomNumber}`
            const email = `email${randomNumber}@education.gov.uk`
```

Reply:
Made the above use reusable code rather than repeated consts